### ISSUE TITLE
proposed Language page grouping

### ISSUE THREAD

**Issue by tbrowder** [perl6/doc#2160]
Proposed  list. Please comment and modify if able. 
Note:  The pipes ('|') separate TITLE and SUBTITLE of each child page 

**A. MIGRATION GUIDES**

+ Perl 5 to Perl 6 guide - Overview | How do I do what I used to do?

+ Perl 5 to Perl 6 guide - Functions | Builtin functions in Perl 5 to Perl 6

+ Perl 5 to Perl 6 guide - Operators | Operators in Perl 5 to Perl 6

+ Perl 5 to Perl 6 guide - Syntax | Syntactic differences between Perl 5 and Perl 6

+ Perl 5 to Perl 6 guide - Special Variables | Special variables in Perl 5 to Perl 6

+ Perl 5 to Perl 6 guide - Nutshell | Technical reference to Perl 6 for Perl 5 users

+ Haskell to Perl 6 - Nutshell | Learning Perl 6 from Haskell, in a nutshell: What do I already know?

+ Javascript (Node) to Perl 6 - Nutshell | Learning Perl 6 from Node.js, in a nutshell: What do I already know?

+ Python to Perl 6 - Nutshell | Learning Perl 6 from Python, in a nutshell: What do I already know?

+ Ruby to Perl 6 - Nutshell | Learning Perl 6 from Ruby, in a nutshell: What do I already know?

**B. TUTORIALS**

+ Contexts and contextualizers | What are contexts and how to get into them

+ Core modules | Core modules that may be useful to module authors

+ Doing math with Perl 6 | Different mathematical paradigms and how they are implemented in this language

+ Entering Unicode characters | Input methods for Unicode characters in editors and the shell

+ Grammar tutorial | An introduction to grammars

+ Input/Output | File-related operations

+ Inter-process communication | Programs running other programs and communicating with them

+ Iterating | Features available for visiting all items in a complex data structure

+ Module development utilities | What can help you write/test/improve your modules

+ Module packages | Creating module packages for code reuse

+ Modules | How to create, use and distribute Perl 6 modules

+ Traps to avoid | Traps to avoid when getting started with Perl 6

**C. GENERAL REFERENCE**

+ About the docs | Meta-documentation

+ Community | Information about the people working on and using Perl 6

+ FAQ | Frequently asked questions about Perl 6

+ Glossary | Glossary of Perl 6 terminology

+ Perl 6 Pod | An easy-to-use markup language

+ Pod 6 tables | The Good, the Bad and the Ugly

+ Terms | Perl 6 terms

+ Testing | Writing and running tests in Perl 6

**D. PERL 6 FUNDAMENTAL TOPICS**

+ Concurrency | Concurrency and asynchronous programming

+ Containers | A low-level explanation of Perl 6 containers

+ Control flow | Statements used to control the flow of execution

+ Data structures | How Perl 6 deals with data structures and what we can expect from them

+ Date and time functions | Processing date and time in Perl 6

+ Enumeration | Using the enum type

+ Exceptions | Using exceptions in Perl 6

+ Functions | Functions and functional programming in Perl 6

+ Grammars | Parsing and interpreting text

+ Hashes and maps | Working with associative arrays/dictionaries/hashes

+ Input/Output The Definitive Guide | Correctly use Perl 6 I/O

+ Lists, Sequences, and Arrays | Positional data constructs

+ Meta-Object Protocol | Introspection and the Perl 6 object system

+ Native calling interface | Call into dynamic libraries that follow the C calling convention

+ Newline handling in Perl 6 | How the different newline characters are handled, and how to change the behavior

+ Numerics | Numeric types available in Perl 6

+ Object orientation | Object orientation in Perl 6

+ Operators | Common Perl 6 infixes, prefixes, postfixes, and more!

+ Packages | Organizing and referencing namespaced program elements

+ Performance | Measuring and improving runtime or compile-time performance

+ Perl 6 native types | Using the types the compiler and hardware make available to you

+ Phasers | Program execution phases and corresponding phaser blocks

+ Pragmas | Special modules for special use

+ Quoting constructs | Writing strings, word lists, and regexes in Perl 6

+ Regexes | Pattern matching against strings

+ Routines | Common Perl 6 routines that are defined outside any class

+ Sets, Bags, and Mixes | Unordered collections of unique and weighted objects in Perl 6

+ Subscripts | Accessing data structure elements by index or key

+ Syntax | General rules of Perl 6 syntax

+ System interaction | Working with the underlying operating system and running applications

+ Traits | Compile-time specification of behavior made easy

+ Type system | Introduction to the type system of Perl 6

+ Unicode | Unicode support in Perl 6

+ Unicode versus ASCII symbols | Unicode symbols and their ASCII equivalents

+ Variables | Variables in Perl 6

**E. PERL 6 ADVANCED TOPICS**

+ Experimental features | New features for brave users

**Comment by AlexDaniel** [comment]
I think that's a great idea. Maybe see also https://github.com/perl6/doc/issues/114 because it also groups pages by category, but the structure proposed here is just fine I think.

**Comment by tbrowder** [comment]
I’ll break the Perl 6 specifics up into something like you suggested.

I also suggest, as long as this ordering system is used, that I integrate an improved version of the p6 tool I built to encapsulate the  title data and actual file name for easier modification of the internal sort key.

**Comment by softmoth** [comment]
Tom, I made a few edits, feel free to disregard if you don't agree. The main one is moving Perl5 to Perl6 Nutshell lower, and changing the title to be a bit more imposing. As it states at the top of that page

> Hence this should not be mistaken for a beginner tutorial or a promotional overview of Perl 6; it is intended as a technical reference for Perl 6 learners with a strong Perl 5 background and for anyone porting Perl 5 code

I tried to normalize a few of the subheadings.

There's one typo that I fixed, which you'll want to keep even if you disagree w/ the other changes: The title separator for "Haskell to Perl 6" was a tab rather than a pipe.

I think the other changes are self explanatory. I hope this helps.

**Comment by JJ** [comment]
See also #1969 for a proposal on how to split tutorials...

**Comment by tbrowder** [comment]
Tim, thanks. I just last night made the two p6 categories, and I would appreciate anyone moving the appropriate pages into fundamental or advanced groups.

Also feel free to add a category!  I’m working on a tooling method in the doc tree to make this all fairly easy to modify in the linear layout.

**Comment by stmuk** [comment]
@tbrowder there are tags present in some files which classify as "tutorial"
````
$ grep -l ':tag<tutorial>' *|sort
classtut.pod6
concurrency.pod6
grammar_tutorial.pod6
modules-extra.pod6
performance.pod6
testing.pod6
unicode_entry.pod6
````

**Comment by tbrowder** [comment]
Closed with commit 7afe426.
